test/gtest/uct/ib/test_ib.cc

```diff
@@ -6,19 +6,118 @@
 
 #include <uct/ib/test_ib.h>
 
+test_uct_ib::test_uct_ib() : uct_test(), m_e1(NULL), m_e2(NULL) { }
 
-test_uct_ib::test_uct_ib() : uct_test() {
-    m_e1 = NULL;
-    m_e2 = NULL;
+void test_uct_ib::init() {
+    uct_test::init();
+
+    m_e1 = uct_test::create_entity(0);
+    m_entities.push_back(m_e1);
+
+    m_e2 = uct_test::create_entity(0);
+    m_entities.push_back(m_e2);
+
+    m_e1->connect(0, *m_e2, 0);
+    m_e2->connect(0, *m_e1, 0);
 
+    test_uct_ib::m_ib_am_handler_counter = 0;
+}
+
+void test_uct_ib::cleanup() {
```

Comment:
this is redundant

Reply:
i think you can call test_ib::cleanup() without really defining it (from the places you described), it will default to uct_test::cleanup() if test_ib::cleanup() is not defined